modules/serverless-kn-func-emit.adoc

```diff
@@ -0,0 +1,86 @@
+[id="serverless-kn-func-emit_{context}"]
+= Emitting a test event to a deployed function
+
+You can use the `kn func emit` CLI command to emit a CloudEvent to a function that is either deployed locally or deployed to your {product-title} cluster. This command can be used to test that a function is working and able to receive events correctly.
+
+.Example command
+[source,terminal]
+----
+$ kn func emit
+----
+
+The `kn func emit` command executes on the local directory by default, and assumes that this directory is a function project.
+
+== Optional parameters
+
+You can specify the following optional parameters for the emitted CloudEvent, by using the `kn func emit` CLI command flags:
+
+Event type:: The type of event being emitted. You can find information about the `type` parameter that is set for events from a certain event producer in the documentation for that event producer. For example, the API server source may set the `type` parameter of produced events as `dev.knative.apiserver.resource.update`.
+Event source:: The unique event source that produced the event. This may be a URI for the event source, for example `https://10.96.0.1/`, or the name of the event source.
+Event ID:: A random, unique ID that is created by the event producer.
+Event data:: The data contained in the event. For example, you can specify a `--data` value such as `"Hello world!"` so that the event contains this data string. By default, no data is included in the events created by `kn func emit`.
++
+[NOTE]
+====
+If you are sending events from an existing event source, the event source might already specify data for produced events.
```

Comment:
I'm not sure I understand the intent of this note. The `kn func emit` command does not send from an existing source. I think there is value in mentioning existing event sources, but I wonder if this could be expanded a bit to be a little clearer. This is just a suggestion. 

```suggestion
Functions that have been deployed to a cluster may be responding to events from an existing event source that provides values for properties such as `source` and `type` noted above. These events often have a `data` value in JSON form, that captures the domain specific context of the event. For example, an event sent from Apache Camel may have a `type` of `org.apache.camel.event` and a `data` property containing data associated with the Camel event. Using the CLI flags noted in this document, developers can simulate those events for local testing.
```

Reply:
Sounds good, thanks. I did remove the bit about camel though since camel sources are deprecated now afaik.